paymentsheet/src/main/java/com/stripe/android/paymentelement/embedded/form/FormActivityViewModel.kt

```diff
@@ -0,0 +1,129 @@
+package com.stripe.android.paymentelement.embedded.form
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.createSavedStateHandle
+import androidx.lifecycle.viewModelScope
+import androidx.lifecycle.viewmodel.CreationExtras
+import com.stripe.android.core.utils.requireApplication
+import com.stripe.android.lpmfoundations.luxe.isSaveForFutureUseValueChangeable
+import com.stripe.android.lpmfoundations.paymentmethod.PaymentMethodMetadata
+import com.stripe.android.model.PaymentIntent
+import com.stripe.android.model.PaymentMethod
+import com.stripe.android.model.PaymentMethodCode
+import com.stripe.android.paymentelement.embedded.EmbeddedFormHelperFactory
+import com.stripe.android.paymentelement.embedded.EmbeddedSelectionHolder
+import com.stripe.android.payments.bankaccount.CollectBankAccountLauncher.Companion.HOSTED_SURFACE_PAYMENT_ELEMENT
+import com.stripe.android.paymentsheet.analytics.EventReporter
+import com.stripe.android.paymentsheet.model.PaymentSelection
+import com.stripe.android.paymentsheet.paymentdatacollection.ach.USBankAccountFormArguments
+import com.stripe.android.paymentsheet.verticalmode.BankFormInteractor
+import com.stripe.android.paymentsheet.verticalmode.DefaultVerticalModeFormInteractor
+import com.stripe.android.paymentsheet.verticalmode.PaymentMethodIncentiveInteractor
+import com.stripe.android.uicore.utils.stateFlowOf
+import javax.inject.Inject
+
+internal class FormActivityViewModel @Inject constructor(
+    paymentMethodMetadata: PaymentMethodMetadata,
+    selectedPaymentMethodCode: PaymentMethodCode,
+    private val selectionHolder: EmbeddedSelectionHolder,
+    private val embeddedFormHelperFactory: EmbeddedFormHelperFactory,
+    val eventReporter: EventReporter
```

Comment:
I've got an idea on how to remove the need for this. I'm going to attempt it in Manage first, and can follow up here if it's successful.